exercises/practice/collatz-conjecture/tests/collatz_conjecture_test.bal

```diff
@@ -0,0 +1,40 @@
+import ballerina/test;
+
+@test:Config {
+    dataProvider: happyCases
+}
+function testCollatzOK(int number, int expected) returns error? {
+    test:assertEquals(collatzSteps(number), expected);
+}
+
+function happyCases() returns map<[int, int]>|error {
+    // a tuple of: [input value, expected result]
+    map<[int, int]> cases = {
+        "zero steps for one": [1, 0],
+        "divide if even": [16, 4],
+        "even and odd steps": [12, 9],
+        "large number of even and odd steps": [1000000, 152]
+    };
+    return cases;
+}
+
+@test:Config {
+    dataProvider: errorCases
+}
```

Comment:
Would I annoy you if I mention parameterized tests vs non-parameterized tests? :)